tests/unit/test_model.py

```diff
@@ -238,3 +239,66 @@ def test_anthropic_model_structured_output_fallback_to_instructor() -> None:
             schema=StructuredOutputTestModel,
         )
         mock_instructor.return_value.chat.completions.create.assert_called_once()
+
+
+def test_langchain_generative_model_redis_cache_setup(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test that Redis cache is set up correctly."""
+    base_chat_model = MagicMock(spec=BaseChatModel)
+    mock_set = MagicMock()
+    monkeypatch.setattr("portia.model.set_llm_cache", mock_set)
+
+    LangChainGenerativeModel(
+        client=base_chat_model,
+        model_name="test",
+    )
+    mock_set.assert_not_called()
+
+    LangChainGenerativeModel(
+        client=base_chat_model,
+        model_name="test",
+        cache=MagicMock(),
+    )
+
+    mock_set.assert_called_once()
+
+
+def test_instructor_manual_cache(monkeypatch: pytest.MonkeyPatch) -> None:
+    """LLM responses are cached when redis URL provided."""
+
+    class DummyModel(BaseModel):
+        pass
+
+    mock_instructor_client = MagicMock()
+    monkeypatch.setattr(
+        "portia.model.instructor.from_openai",
+        MagicMock(return_value=mock_instructor_client),
+    )
+    mock_create = MagicMock(return_value=DummyModel())
+    mock_instructor_client.chat.completions.create = mock_create
+
+    cache = MagicMock()
```

Comment:
```suggestion
    cache = MagicMock(spec=BaseCache)
```
helps validate that the calls made match the interface

Reply:
Nice - done 👍